src/sources/core.js

```diff
@@ -29,15 +35,52 @@ class CoreSource extends Source {
     return new CoreDataset(this, pathParts);
   }
 
-  // The computation of these globals should move here.
+  // The computation of these globals should move here. 
   secondTreeOptions(path) {
     return (global.availableDatasets.secondTreeOptions[this.name] || {})[path] || [];
   }
 
+  async collectResources() {
+    if (!this._allResources) this._allResources = new Map();
+    const s3objects = await parseInventory({...this.inventoryLocation, name: this.name});
+    if (!s3objects.length) {
+      // potentially due to a (transient) failure to parse the inventory -- so we want
+      // to keep any previously set resources
+      return;
+    }
+    /* Comment (to drop before merge)
+    This conditional could be replaced by overwriting `collectResources` in CoreStagingSource.
+    That would require most of the logic in this function to be pulled out into a utility function,
+    which may not be a bad idea */
+    const CollectedResources = this.name==='core' ?
+      CoreCollectedResources : CoreStagingCollectedResources;
+    const datasets = new Map(), files = new Map();
+    s3objects.forEach((object) => {
+      const [name, resourceType] = CollectedResources.objectName(object);
```

Comment:
Grouping of resources is done by those which have the same _name_, so there are decisions to be made around what name to give things. For instance, I think it's clear that we should group datestamped-datasets with their non-datestamped selves. We should probably also group `.zst` and `.gz` files together (although here I don't). Would be great to have others' thoughts.